scripts/globals/spells/dia.lua

```diff
@@ -46,15 +46,19 @@ spell_object.onSpellCast = function(caster, target, spell)
     -- Check for Bio
     local bio = target:getStatusEffect(xi.effect.BIO)
 
-    -- Do it!
-    target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
-    spell:setMsg(xi.msg.basic.MAGIC_DMG)
-
     -- Try to kill same tier Bio (non-default behavior)
     if BIO_OVERWRITE == 1 and bio ~= nil then
         if bio:getPower() == 1 then
             target:delStatusEffect(xi.effect.BIO)
         end
+    else
+        -- Do it!
+        if bio ~= nil then -- if mob has bio effect
+            spell:setMsg(xi.msg.basic.MAGIC_DMG) -- hit for initial damage, but no dot effects
+        else -- otherwise hit for initial damage and add dot
+            target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
+            spell:setMsg(xi.msg.basic.MAGIC_DMG)
+        end
```

Comment:
This probably applies to all.
I would move the setmsg line ouside of the if/else check.
that would make the bio check empty, so, that would have to be edited aswell.

So maybe something like...

if bio == nil then
    target:addStatusEffect(xi.effect.DIA, 1 + dotBonus, 3, duration, 0, 10, 1)
end
spell:setMsg(xi.msg.basic.MAGIC_DMG)

Reply:
changed this to clean things up